src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,862 @@
+import { Card, CardBlock, CardTitle } from "@dellstorage/clarity-react/cards";
+import { Table } from "@dellstorage/clarity-react/tables";
+import FilterPanel from "components/DataGridWithInfiniteScroll/FilterPanel";
+import filter from "assets/images/filter.svg";
+import filterOpen from "assets/images/filter-solid.svg";
+import emptyDatagrid from "assets/images/emptyDatagrid.svg";
+import InfiniteScroll from "react-infinite-scroll-component";
+import {
+    Spinner,
+    SpinnerSize,
+} from "@dellstorage/clarity-react/spinner/Spinner";
+import { Constants } from "components/DataGridWithInfiniteScroll/Constants";
+
+/**
+ * General component description :
+ * DataGridWithInfiniteScroll :
+ * Datagrids are for organizing large volumes of data that users can perform actions on.
+ */
+
+
+/**
+ * Enum for RowTpye :
+ * @param {EXPANDABLE} for enabling expandable rows
+ * @param {COMPACT} for enabling compact rows
+ * @param {ROWS_WITH_DETAIL_PANE} for enabling detail pane for rows
+ * @param {EXPANDABLE_ROWS_WITH_DETAIL_PANE} for enabling detail pane for expandable rows
+ * @param {COMPACT_ROWS_WITH_DETAIL_PANE} for enabling detail pane for compact rows
+ */
+export enum GridRowType {
+    EXPANDABLE = "expandable",
+    COMPACT = "compact",
+    ROWS_WITH_DETAIL_PANE = "rows_with_detail_pane",
+    EXPANDABLE_ROWS_WITH_DETAIL_PANE = "expandable_rows_with_detail_pane",
+    COMPACT_ROWS_WITH_DETAIL_PANE = "compact_rows_with_detail_panes",
+}
+
+/**
+ * Enum for GridSelectionType :
+ * @param {MULTI} for enabling multi row select
+ * @param {SINGLE} for enabling single row select
+ */
+export enum GridSelectionType {
+    MULTI = "multi",
+    SINGLE = "single",
+}
+
+/**
+ * type for DataGridRow :
+ * @param {rowData} row data
+ * @param {rowID} unique ID to identify row
+ * @param {isSelected} set to true if row is selected
+ * @param {className} CSS class name
+ * @param {style} CSS style
+ * @param {expandableRowData} Expandable row content
+ */
+
+export type DataGridRow = {
+    className?: string;
+    style?: any;
+    rowID?: number; // not to take from user
+    isSelected?: boolean;
+    disableRowSelection?: boolean;
+    expandableRowData?: ExpandableRowDetails;
+};
+/**
+ * type for datagrid expandable row data :
+ * @param {isLoading} if true then show loading icon for expandable row
+ * @param {onRowExpand} callback function to  fetch expandable row contents
+ * @param {onRowContract} callback function for additional logic after row contracts
+ * @param {expandableContent} content to show after row expand
+ * @param {isExpanded} true if row is already expanded
+ * @param {hideRowExpandIcon} if true then hide icon for expandable row
+ */
+export type ExpandableRowDetails = {
+    isLoading?: boolean;
+    onRowExpand?: (row: DataGridRow) => Promise<any>;
+    onRowContract?: (row: DataGridRow) => void;
+/**
+ * TODO:
+ * Type 'Any' will be changed to specific types while writing wrappers for props.
+ */
+    expandableContent?: any;
```

Comment:
```suggestion
    expandableContent?: any; // TODO : Change "any" to specific type while writing wrappers 
```
Please add inline comment